tests/e2e/greptimedbcluster/test_cluster_enable_monitoring.go

```diff
@@ -0,0 +1,177 @@
+// limitations under the License.
+
+package greptimedbcluster
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sort"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/jackc/pgx/v5"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	greptimev1alpha1 "github.com/GreptimeTeam/greptimedb-operator/apis/v1alpha1"
+	"github.com/GreptimeTeam/greptimedb-operator/controllers/common"
+	"github.com/GreptimeTeam/greptimedb-operator/tests/e2e/helper"
+)
+
+// TestClusterEnableMonitoring tests a cluster that enables monitoring.
+func TestClusterEnableMonitoring(ctx context.Context, h *helper.Helper) {
+	const (
+		testCRFile  = "./testdata/resources/cluster/enable-monitoring/cluster.yaml"
+		testSQLFile = "./testdata/sql/cluster/partition.sql"
+	)
+
+	By(fmt.Sprintf("greptimecluster test with CR file %s and SQL file %s", testCRFile, testSQLFile))
+
+	testCluster := new(greptimev1alpha1.GreptimeDBCluster)
+	err := h.LoadCR(testCRFile, testCluster)
+	Expect(err).NotTo(HaveOccurred(), "failed to load greptimedbcluster yaml file")
+
+	err = h.Create(ctx, testCluster)
+	Expect(err).NotTo(HaveOccurred(), "failed to create greptimedbcluster")
+
+	By("Check the status of testCluster")
+	Eventually(func() error {
+		clusterPhase, err := h.GetPhase(ctx, testCluster.Namespace, testCluster.Name, new(greptimev1alpha1.GreptimeDBCluster))
+		if err != nil {
+			return err
+		}
+
+		if clusterPhase != greptimev1alpha1.PhaseRunning {
+			return fmt.Errorf("cluster is not running")
+		}
+
+		return nil
+	}, helper.DefaultTimeout, time.Second).ShouldNot(HaveOccurred())
+
+	err = h.Get(ctx, client.ObjectKey{Name: testCluster.Name, Namespace: testCluster.Namespace}, testCluster)
+	Expect(err).NotTo(HaveOccurred(), "failed to get cluster")
+
+	By("Execute distributed SQL test")
+	frontendAddr, err := h.PortForward(ctx, testCluster.Namespace, common.ResourceName(testCluster.Name, greptimev1alpha1.FrontendComponentKind), int(testCluster.Spec.PostgreSQLPort))
+	Expect(err).NotTo(HaveOccurred(), "failed to port forward frontend service")
+	Eventually(func() error {
+		conn, err := net.Dial("tcp", frontendAddr)
+		if err != nil {
+			return err
+		}
+		conn.Close()
+		return nil
+	}, helper.DefaultTimeout, time.Second).ShouldNot(HaveOccurred())
+
+	err = h.RunSQLTest(ctx, frontendAddr, testSQLFile)
+	Expect(err).NotTo(HaveOccurred(), "failed to run sql test")
+
+	monitoringAddr, err := h.PortForward(ctx, testCluster.Namespace, common.ResourceName(common.MonitoringServiceName(testCluster.Name), greptimev1alpha1.StandaloneKind), int(testCluster.Spec.PostgreSQLPort))
+	Expect(err).NotTo(HaveOccurred(), "failed to port forward monitoring service")
+	Eventually(func() error {
+		conn, err := net.Dial("tcp", monitoringAddr)
+		if err != nil {
+			return err
+		}
+		conn.Close()
+		return nil
+	}, helper.DefaultTimeout, time.Second).ShouldNot(HaveOccurred())
+	err = testMonitoringStandalone(ctx, monitoringAddr)
```

Comment:
It should be test of greptimedbcluster ?

Reply:
We need to check the collected data, which is stored in the standalone.